File: app/src/main/java/com/supercilex/robotscouter/ui/teamlist/DeleteTeamDialog.java
package com.supercilex.robotscouter.ui.teamlist;

import android.app.AlertDialog;
import android.app.Dialog;
import android.content.DialogInterface;
import android.os.Bundle;
import android.os.Parcelable;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.app.DialogFragment;
import android.support.v4.app.FragmentManager;

import com.supercilex.robotscouter.R;
import com.supercilex.robotscouter.data.util.TeamHelper;
import com.supercilex.robotscouter.util.Constants;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class DeleteTeamDialog extends DialogFragment implements AlertDialog.OnClickListener {
    private static final String TAG = "DeleteTeamDialog";
    private static final String TEAMS_KEY = "teams_key";

    private List<TeamHelper> mTeamHelpers = new ArrayList<>();

    public static void show(FragmentManager manager, List<TeamHelper> teams) {
        DialogFragment dialog = new DeleteTeamDialog();

        Bundle args = new Bundle();
        args.putParcelableArray(TEAMS_KEY, teams.toArray(new TeamHelper[teams.size()]));
        dialog.setArguments(args);

        dialog.show(manager, TAG);
    }

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        for (Parcelable parcelable : getArguments().getParcelableArray(TEAMS_KEY)) {
            mTeamHelpers.add((TeamHelper) parcelable);
        }
        Collections.sort(mTeamHelpers);
    }

    @NonNull
    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        StringBuilder deletedTeams = new StringBuilder();
        for (int i = 0; i < mTeamHelpers.size(); i++) {
            deletedTeams.append(i + 1)
                    .append(". ")
                    .append(mTeamHelpers.get(i))
                    .append('\n');
        }

        return new AlertDialog.Builder(getContext())
                .setTitle(R.string.confirm_action)
                .setMessage(mTeamHelpers.size() == Constants.SINGLE_ITEM
                                    ? null : getString(R.string.caution_delete, deletedTeams))
                .setPositiveButton(R.string.delete, this)
                .setNegativeButton(android.R.string.no, null)
                .create();
    }

    @Override
    public void onClick(DialogInterface dialog, int which) {
        for (TeamHelper teamHelper : mTeamHelpers) {
            teamHelper.deleteTeam();
        }
    }
}

File: app/src/main/java/com/supercilex/robotscouter/ui/teamlist/DeleteTeamDialog.kt
package com.supercilex.robotscouter.ui.teamlist

import android.app.Dialog
import android.os.Bundle
import android.support.v4.app.FragmentManager
import android.text.SpannableStringBuilder
import android.view.View
import androidx.core.text.bold
import com.supercilex.robotscouter.R
import com.supercilex.robotscouter.data.model.Team
import com.supercilex.robotscouter.util.data.getTeamList
import com.supercilex.robotscouter.util.data.model.trash
import com.supercilex.robotscouter.util.data.toBundle
import com.supercilex.robotscouter.util.isSingleton
import com.supercilex.robotscouter.util.ui.BottomSheetDialogFragmentBase
import com.supercilex.robotscouter.util.ui.show
import com.supercilex.robotscouter.util.unsafeLazy
import kotlinx.android.synthetic.main.dialog_delete_team.*

class DeleteTeamDialog : BottomSheetDialogFragmentBase(), View.OnClickListener {
    override val containerView: View by unsafeLazy {
        View.inflate(context, R.layout.dialog_delete_team, null)
    }
    private val teams: List<Team> by unsafeLazy { arguments!!.getTeamList().sorted() }

    override fun onDialogCreated(dialog: Dialog, savedInstanceState: Bundle?) {
        message.text = run {
            val message = resources.getQuantityString(
                    R.plurals.team_delete_message,
                    teams.size,
                    teams.first()
            )

            if (teams.isSingleton) return@run message

            val deletedTeams = SpannableStringBuilder(message).append('\n')
            for ((i, team) in teams.withIndex()) {
                deletedTeams.bold {
                    append('\n').append("${i + 1}. ")
                }.append(team.toString())
            }
            deletedTeams
        }
        delete.setOnClickListener(this@DeleteTeamDialog)
    }

    override fun onClick(v: View) {
        for (team in teams) team.trash()
        dismiss()
    }

    companion object {
        private const val TAG = "DeleteTeamDialog"

        fun show(manager: FragmentManager, teams: List<Team>) =
                DeleteTeamDialog().show(manager, TAG, teams.toBundle())
    }
}
